slackbot/dispatcher.py

```diff
@@ -205,6 +203,35 @@ def react(self, emojiname):
             channel=self._body['channel'],
             timestamp=self._body['ts'])
 
+    def user_list(self):
+        return self._client.webapi.users.list()
+
+    def name_to_id(self, name):
+        for x in self._client.users.keys():
+            if self._client.users[x]['name'] == name:
+                return self._client.users[x]['id']
+        return "Not Found"
+
+    def id_to_name(self, id):
+        for x in self._client.users.keys():
+            if self._client.users[x]['id'] == id:
+                return self._client.users[x]['name']
+        return "Not Found"
+
+    def sent_by(self):
+        for x in self._client.users.keys():
+            if self._client.users[x]['id'] == self.body['user']:
+                return self._client.users[x]['name']
+        return "Not Found"
+
+    def upload_snippet(self, file_text, title="File"):
+        with create_tmp_file() as f:
+                with open(f, 'w') as tmpf:
+                    tmpf.write(file_text)
+                    tmpf.flush()
+                    tmpf.close()
+                self.channel.upload_file(title, f)
+
```

Comment:
is upload_snippet() something generic enough that it should belong in the core here? (it might be)